src/python/pants/engine/internals/docker_test.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+import json
+import subprocess
+
+import pytest
+
+from pants.engine.internals.docker import DockerResolveImageRequest, DockerResolveImageResult
+from pants.engine.platform import Platform
+from pants.engine.rules import QueryRule
+from pants.testutil.rule_runner import RuleRunner
+
+
+@pytest.fixture
+def rule_runner() -> RuleRunner:
+    return RuleRunner(rules=[QueryRule(DockerResolveImageResult, (DockerResolveImageRequest,))])
+
+
+def test_resolve_image_id(rule_runner: RuleRunner) -> None:
+    platform = Platform.create_for_localhost()
```

Comment:
This test fails on MacOS (at least on M1) with `no matching manifest for darwin/arm64 in the manifest list entries`

Which makes sense. Should we be hardcoding a linux platform here, or skipping the test on macos?